calculadora.py

```diff
@@ -1,46 +1,57 @@
-    return x - y
-
-# Função principal
-def calculadora ():
-    num1 = int(input("Digite o primeiro número: "))
-    num2 = int(input("Digite o segundo número: "))
-    result = subtrair(num1, num2)
-    print("resultado: "+ str(result))
-
-calculadora()
-def multiplicar (x,y):
-    return x * y
-
-# Função principal
-def calculadora ():
-    num1 = int(input("Digite o primeiro número: "))
-    num2 = int(input("Digite o segundo número: "))
-    result = multiplicar (num1, num2)
-    print("resultado: "+ str(result))
-
-calculadora()
-# Função para dividir
-def dividir (x,y):
-    return x / y
-
-# Função principal
-def calculadora ():
-    num1 = int(input("Digite o primeiro número: "))
-    num2 = int(input("Digite o segundo número: "))
-    result = dividir(num1, num2)
-    print("resultado: "+ str(result))
+def calculadora():
+    lista_historico = []  # Inicializa a lista para o histórico
+
+        #Escolher operacao para ser efetuada pela calculadora
+    while True:
+        op = input("Digite a operação (soma, subtracao, multi, div, hist ou sair): ").strip().lower()
+        if op not in ["soma", "hist", "subtracao", "div", "multi", "sair"]:
+            print("operacao invalida")
+            continue
+
+        if op == "sair":
+            break
+        if op != "hist":
+        #Após escolher operacao escolher dois numeros para ser feita a conta
+            num1 = int(input("Digite o primeiro número: "))
+            num2 = int(input("Digite o segundo número: "))
+            
+        #Efetua a operacao de soma com os dois numeros escolhidos e mostra a conta por extenso
+        if op == "soma":
+            result = num1 + num2
+            conta = str(num1) + " + " + str(num2) + " = " + str(result)
+            print(conta)
+            lista_historico.append(conta)
+            
+        #Efetua a operacao de multiplicacao com os dois numeros escolhidos e mostra a conta por extenso
+        elif op == "multi":
+            result = num1 * num2
+            conta = str(num1) + " * " + str(num2) + " = " + str(result)
+            print(conta)
+            lista_historico.append(conta)
+            
+        #Efetua a operacao de subtracao com os dois numeros escolhidos e mostra a conta por extenso
+        elif op == "subtracao":
+            result = num1 - num2
+            conta = str(num1) + " - " + str(num2) + " = " + str(result)
+            print(conta)
+            lista_historico.append(conta)
+            
+       #Efetua a operacao de divisao com os dois numeros escolhidos e mostra a conta por extenso
+        elif op == "div":
+            result = num1 / num2
+            conta = str(num1) + " / " + str(num2) + " = " + str(result)
+            print(conta)
+            lista_historico.append(conta)
+            
+         #Mostra o historico da calculadora
+        elif op == "hist":
+            print("Histórico das últimas 5 operações:")
+            for i, operacao in enumerate(lista_historico[-5:], 1):
```

Comment:
demasiado complexo... com a linha 53 e 54 já manténs apenas as últimas 5 operações pelo que aqui bastaria um for simples

`for conta in lista_historico:
       print(conta)`